server/src/session.ts

```diff
@@ -111,18 +113,48 @@ export class Session {
     conn.onReferences(p => this.onReferences(p));
     conn.onHover(p => this.onHover(p));
     conn.onCompletion(p => this.onCompletion(p));
-    conn.onNotification(notification.NgccComplete, p => this.handleNgccNotification(p));
   }
 
-  private handleNgccNotification(params: notification.NgccCompleteParams) {
-    const {configFilePath} = params;
-    if (!params.success) {
+  private async runNgcc(configFilePath: string) {
+    this.connection.sendProgress(NgccProgressType, NgccProgressToken, {
+      done: false,
+      configFilePath,
+      message: `Running ngcc for project ${configFilePath}`,
+    });
+
+    let success = false;
+
+    try {
+      await resolveAndRunNgcc(configFilePath, {
+        report: (msg: string) => {
+          this.connection.sendProgress(NgccProgressType, NgccProgressToken, {
+            done: false,
+            configFilePath,
+            message: msg,
+          });
+        },
+      });
+      success = true;
+    } catch (e) {
       this.error(
           `Failed to run ngcc for ${configFilePath}:\n` +
-          `${params.error}\n` +
-          `Language service will remain disabled.`);
+          `    ${e.message}\n` +
+          `    Language service will remain disabled.`);
+    } finally {
+      this.connection.sendProgress(NgccProgressType, NgccProgressToken, {
+        done: true,
+        configFilePath,
```

Comment:
Does this mean the client isn't really notified when an error happens? It would just show up in server logs and the client would see that ngcc is 'done'?